### PR TITLE
[GPU] Move local_block_io_support check to kernel selector helper

### PR DESCRIPTION
**Pull request by vladimir-paramuzov**
### Details:
 - `is_local_block_io_supported` moved to kernel selector helper

### CODE REVIEW

File: src/plugins/intel_gpu/src/graph/impls/ocl/kernel_selector_helper.h [hunk 1]
```diff
@@ -226,12 +140,10 @@ inline params_t get_default_params(const kernel_impl_params& param_info) {
         size_t op_id = 0;
         for (auto& fused_prim : param_info.fused_desc) {
             kernel_selector::fused_operation_desc desc;
-            desc.op_params = std::move(fused_prim.f_param);
+            desc.op_params = convert_fuse_params(fused_prim.f_param);
+
+            OPENVINO_ASSERT(desc.op_params != nullptr, "[GPU] Invalid fused operation (", param_info.desc->id , ") of type ", param_info.desc->type_string());
```
Comment (sshlyapn) [inline src/plugins/intel_gpu/src/graph/impls/ocl/kernel_selector_helper.h:145]: nit: looks like `convert_fuse_params` will never return nullptr

**Review by sshlyapn — APPROVED** [review]
Looks good to me, just left some minor comments

File: src/plugins/intel_gpu/src/graph/impls/ocl/kernel_selector_helper.cpp [hunk 2]
```diff
@@ -1160,7 +1217,7 @@ void set_params(const kernel_impl_params& param_info, kernel_selector::params& p
     params.engineInfo.enable_sub_groups_emulation = true;
     params.engineInfo.bOptHintsSupport = false;
 
-    params.engineInfo.bLocalBlockIOSupport = device_info.supports_local_block_io && program->is_local_block_io_supported();
+    params.engineInfo.bLocalBlockIOSupport = query_local_block_io_supported(engine, config);
```
Comment (yeonbok) [inline src/plugins/intel_gpu/src/graph/impls/ocl/kernel_selector_helper.cpp:1220]: If we query this here, we'll query every time of update_impl in dynamic shape which will affect the performance.. wont't it?

Comment (vladimir-paramuzov) [reply src/plugins/intel_gpu/src/graph/impls/ocl/kernel_selector_helper.cpp:1220]: I don't expect visible perf impact as starting from second set_params call result will be returned from cache, so it's just a matter of getting element from map

**Pull request merged**